src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+@Slf4j
+public class OrgServiceImpl implements OrgService {
+
+    @Autowired
+    private PayloadValidation payloadValidation;
+
+    private @Autowired CassandraOperation cassandraOperation;
+
+    private @Autowired OutboundRequestHandlerServiceImpl outboundRequestHandlerServiceImpl;
+
+    private @Autowired CbServerProperties cbServerProperties;
+
+    private @Autowired AccessTokenValidator accessTokenValidator;
+
+    private @Autowired DemandService demandService;
+
+    @Override
+    public ApiResponse readFramework(String frameworkName, String orgId, String userAuthToken) {
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_ORG_CREATE);
+        try {
+            if (frameworkName == null || orgId == null) {
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg("OrgID and FrameworkId is Missing");
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            String userId = accessTokenValidator.verifyUserToken(userAuthToken);
+            if (StringUtils.isBlank(userId)) {
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg(Constants.USER_ID_DOESNT_EXIST);
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            if (!demandService.isSpvRequest(userId,Constants.MDO_ADMIN)) {
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg("User does not have the required role:");
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            Map<String, Object> propertyMap = new HashMap<>();
+            propertyMap.put(Constants.ID, orgId);
+            List<Map<String, Object>> orgDetails = cassandraOperation.getRecordsByPropertiesWithoutFiltering(Constants.KEYSPACE_SUNBIRD, Constants.ORG_TABLE, propertyMap, null, 1);
+            if (orgDetails.isEmpty()) {
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg("Organization not found");
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            String fwName = (String) orgDetails.get(0).get(Constants.FRAMEWORKID);
+            if (fwName == null) {
+                StringBuilder strUrl = new StringBuilder(cbServerProperties.getKnowledgeMS());
+                strUrl.append(cbServerProperties.getOdcsFrameworkCreate());
+                Map<String, Object> createReq = createFrameworkRequest(orgId, frameworkName);
```

Comment:
Is this means you need to create a framwork in read request only

Reply:
yes if it is not there we need to create.